add support for getting CA info from the server

redegade opened the pull request:
fabric-ca-client should support the CA server's /cainfo api endpoint

redegade committed:
add support for getting CA info from the server

Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

redegade committed:
fixed lint issue

Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

nadongjun committed:
Fix some wrong spelling and url in the README.md (#434)

Signed-off-by: nadongjun <aa9814526@gmail.com>
Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

bestbeforetoday reviewed (changes requested):
Thank you for the contribution. I appreciate you putting in the effort to provide good unit tests. Just a couple of additions that I think would be worthwhile:
1. A test for the value returned from FabricCAClient.getCaInfo(). You might achieve this by stubbing/mocking only the POST call (made by FabricCAClient) in your test of FabricCAServices.getCAInfo(), but up to you which approach you prefer
2. A (very) simple integration test case added to `test/integration/fabric-ca-services-tests.js` to give confidence that the code really works at runtime, and doesn't get broken by future changes

denyeart committed:
Update "master" branch references to "main"

Update "master" branch references to "main".

Signed-off-by: David Enyeart <enyeart@us.ibm.com>
Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

Commit:
[FABN-1711] prune deprecated fabric-client (#449)

May will be EOL of 1.4 LTS

Signed-off-by: DavidLiu <david.yx.liu@oracle.com>
Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

bestbeforetoday committed:
Update dependencies to address security vulnerabilities

Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>
Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

bestbeforetoday committed:
FABN-1714: Allow transactions to be serialized (#457)

Transaction state can be serialized using Transaction.serialize(), and later reconstructed using Contract.deserializeTransaction(). This allows a transaction to be persisted and for the same transaction (including nonce and transaction ID) to be submitted again following a client application restart.

Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>
Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

bestbeforetoday committed:
FABN-1714: Exclude wallet files from license scan (#458)

Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>
Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

bestbeforetoday committed:
FABN-1714: Fix JSDoc generation of Transaction class (#460)

Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>
Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

bestbeforetoday committed:
FABN-1715: Always check all files for license headers (#465)

Previously the list of files to check was obtained by diffing the last git commit, which produced an empty list in PR builds within Azure Pipelines and allowed PR builds to pass with missing license header.

Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>
Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

davidkel committed:
[FABN-1710] Fix HSM Persistence issues (#464)

* Merged Pull request #450 into main
* Extend HSM tests

Signed-off-by: D <d_kelsey@uk.ibm.com>
Co-authored-by: Angel Kafazov <akafazov@cst-bg.net>
Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

davidkel committed:
[FABN-1554] Add label support for HSM (#468)

Add the option to specify label instead of slot
If both are provided then slot will be ignored

Signed-off-by: D <d_kelsey@uk.ibm.com>
Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

bestbeforetoday committed:
FABN-1716: Fix hang on application exit (#469)

A change in v1.2.0 and later versions of @grpc/grpc-js is preventing client applications from exiting due to interval timers on the event queue created by @grpc/grpc-js/src/subchannel.ts. This appears to be linked to behaviour when the `grpc.keepalive_permit_without_calls` setting is enabled, which it is by default in the fabric-common configuration. This change disables that setting by default until a fix to @grpc/grpc-js is released.

Additionally, connection timeout Promises during sends to orderers in Committer.js were not being closed correctly. This did not cause a breakage since the promise they rejected was already resolved if the connection was successful, but could leave timers running unnecessarily in the event queue.

Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>
Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

bestbeforetoday committed:
FABN-1716: Update @grpc/grpc-js version (#474)

@grpc/grpc-js v1.3.4 fixes application hang on exit issue in gRPC, so allows the `grpc.keepalive_permit_without_calls` setting to be re-enabled by default.

Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>
Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

denyeart committed:
Reference Asset Transfer samples in SDK docs (#478)

Reference Asset Transfer samples in SDK docs instead of
Fabcar and Commercial Paper.  The Asset Transfer
samples have corresponding beginner tutorials and
Fabric docs and is the recommend place to start now.

Signed-off-by: David Enyeart <enyeart@us.ibm.com>
Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

Commit:
[bug-fix] eventhub status value is now interpreted as http code (#480)

Signed-off-by: DavidLiu <david.yx.liu@oracle.com>
Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

Zzocker committed:
FABN-1724: add support async signing of message (#479)

Signed-off-by: Pritam Singh <pkspritam16@gmail.com>
Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

sapthasurendran committed:
FABN-1657: Retry block event service when no peers available (#481)

Signed-off-by: sapthasurendran <saptha.surendran@ibm.com>
Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

sapthasurendran committed:
FABN-1553: tslint to eslint migration (#484)

Signed-off-by: sapthasurendran <saptha.surendran@ibm.com>
Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

bestbeforetoday committed:
Fix build break

New TypeScript release caused compile / lint failures.

- Pinned to a more specific TypeScript version
- Minor type and lint changes
- Some additional exclusions to license check to avoid failures in development environment

Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>
Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

sapthasurendran committed:
FABN-1530: Timestamp added to full and private block transaction events (#490)

* FABN-1530:timestamp added to full private transactionevents

Signed-off-by: sapthasurendran <saptha.surendran@ibm.com>
Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

sapthasurendran committed:
FABN-1542: Added timeout info in transaction submit err messages (#495)


Signed-off-by: sapthasurendran <saptha.surendran@ibm.com>
Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

sapthasurendran committed:
FABN-1547:JSDoc Cleanup (#496)


Signed-off-by: sapthasurendran <saptha.surendran@ibm.com>
Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

bestbeforetoday commented:
@redegade This PR has been unchanged for quite a while. Are you planning to make the suggested changes so it is ready to be merged?

redegade committed:
added a test for the value returned from getCAInfo

Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

redegade committed:
added simple integration test case

Signed-off-by: Redi Kurti <redikurti.s@gmail.com>

redegade committed:
Merge branch 'hyperledger:main' into master
